tensorflow_probability/python/mcmc/transformed_kernel.py

```diff
@@ -110,6 +106,33 @@ def transformed_log_prob_fn(*state_parts):
   return transformed_log_prob_fn
 
 
+def target_log_prob_getter(kernel):
+  kernel_stack = [kernel]
+  while 'target_log_prob_fn' not in kernel.parameters:
+    if 'inner_kernel' not in kernel.parameters:
+      raise ValueError('"None of the nested `inner_kernel`s contains a '
+                       '`target_log_prob_fn`."')
+    kernel = kernel.inner_kernel
+    kernel_stack.append(kernel)
+  return kernel.parameters['target_log_prob_fn'], kernel_stack
+
+
+def target_log_prob_setter(kernel_stack, new_target_log_prob):
+  # edit the appropriate kernel with the new log_prob_fn
+  last_kernel = kernel_stack.pop()
+  last_kernel_kwargs = last_kernel.parameters.copy()
+  last_kernel_kwargs.update(target_log_prob_fn=new_target_log_prob)
```

Comment:
Let's follow @csuter's offline suggestion and encapsulate this updating logic as a `copy(self, **override_parameters_kwargs)` method on the `TransitionKernel` base class, like in `Distribution` (don't worry about the `batch_slice` stuff in `Distribution`; that's not relevant here).

Reply:
Done in similar style to `Distribution`